apps/src/applab/designMode.js

```diff
@@ -549,6 +548,30 @@ designMode.onDuplicate = function (element, event) {
   return duplicateElement;
 };
 
+function duplicateScreen(element) {
+  let sourceScreen = $(element);
+  let newScreen = designMode.createScreen();
+  designMode.changeScreen(newScreen);
+
+  // Unwrap the draggable wrappers around the elements in the source screen:
+  let madeUndraggable = makeUndraggable(sourceScreen.children());
+
+  // Clone each child of the source screen into the new screen (with new ids):
+  sourceScreen.children().each(function () {
+    var clonedChild = $(this).clone(true)[0];
+    let elementType = elementLibrary.getElementType(clonedChild);
+    elementUtils.setId(clonedChild, elementLibrary.getUnusedElementId(elementType.toLowerCase()));
+    designMode.attachElement(clonedChild);
+  });
+
+  // Restore the draggable wrappers on the elements in the source screen:
+  if (madeUndraggable) {
+    makeDraggable(sourceScreen.children());
+  }
+
+  return newScreen;
+}
```

Comment:
We should have some test coverage somewhere of this new functionality.

Reply:
yes, will do. Just figured out that we can do it in the integration tests. I didn't see design mode tests in unit tests or ui tests.